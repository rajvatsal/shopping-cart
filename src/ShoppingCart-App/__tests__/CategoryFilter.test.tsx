import Categories from '../CategoryFilter.jsx'
import { vi, describe, it, expect } from 'vitest'
import { render } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'

describe('CATEGORY FILTER', () => {
  it('Structure', () => {
    const { getByTestId } = render(
      <Categories add={() => 3} remove={() => 3} />
    )
    const container = getByTestId('categories-component')

    expect(container).toMatchSnapshot()
  })

  it('Checkbox works', async () => {
    const add = vi.fn()
    const remove = vi.fn()
    const user = userEvent.setup()
    const { getByRole } = render(<Categories add={add} remove={remove} />)

    expect(add).toBeCalledTimes(0)
    expect(remove).toBeCalledTimes(0)

    await user.click(getByRole('checkbox', { name: /jewelery/i }))
    expect(add).toBeCalledTimes(1)

    await user.click(getByRole('checkbox', { name: /jewelery/i }))
    expect(remove).toBeCalledTimes(1)

    await user.click(getByRole('checkbox', { name: /electronics/i }))
    expect(add).toBeCalledTimes(2)

    await user.click(getByRole('checkbox', { name: /^men's clothing/i }))
    expect(add).toBeCalledTimes(3)

    await user.click(getByRole('checkbox', { name: /women's clothing/i }))
    expect(add).toBeCalledTimes(4)

    await user.click(getByRole('checkbox', { name: /^men's clothing/i }))
    expect(remove).toBeCalledTimes(2)
  })
})
