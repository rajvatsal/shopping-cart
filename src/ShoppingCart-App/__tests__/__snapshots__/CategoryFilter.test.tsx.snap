// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CATEGORY FILTER > Structure 1`] = `
<div
  data-testid="categories-component"
>
  <h3>
    CATEGORIES
  </h3>
  <div>
    <label>
      <input
        type="checkbox"
      />
      Jewelery
    </label>
    <label>
      <input
        type="checkbox"
      />
      Electronics
    </label>
    <label>
      <input
        type="checkbox"
      />
      Men's Clothing
    </label>
    <label>
      <input
        type="checkbox"
      />
      Women's Clothing
    </label>
  </div>
</div>
`;
